### PR TITLE
raftstore-v2: support unsafe recovery demote

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Issue Number: ref #15108 

What's Changed:

```commit-message
* raftstore-v2: support unsafe recovery demote
* raftstore-v2: do not step force leader on confchange
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]

File: components/raftstore-v2/src/operation/unsafe_recovery/demote.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    pub fn unsafe_recovery_maybe_finish_demote_failed_voters<T: Transport>(
+        &mut self,
+        ctx: &mut StoreContext<EK, ER, T>,
+    ) {
+        let Some(UnsafeRecoveryState::DemoteFailedVoters {
+            syncer,
+            failed_voters,
+            target_index,
+            demote_after_exit,
+        }) = self.unsafe_recovery_state() else {return };
```
Comment (Connor1996) [inline components/raftstore-v2/src/operation/unsafe_recovery/demote.rs:102]:
```suggestion
        }) = self.unsafe_recovery_state() else { return };
```

**Review by Connor1996 — APPROVED** [review]
rest LGTM

**Pull request merged**